src/NuGet.Clients/VisualStudio.Facade/NuGetUIThreadHelper.cs

```diff
@@ -33,28 +33,29 @@ public static JoinableTaskFactory JoinableTaskFactory
         /// Retrieve the CPS enabled JoinableTaskFactory for the current version of Visual Studio.
         /// This overrides the default VsTaskLibraryHelper.ServiceInstance JTF.
         /// </summary>
-        public static void SetJoinableTaskFactoryFromService(IComponentModel componentModel)
+        public static void SetJoinableTaskFactoryFromService(IProjectServiceAccessor projectServiceAccessor)
         {
-            if (componentModel == null)
+            if (projectServiceAccessor == null)
             {
-                throw new ArgumentNullException(nameof(componentModel));
+                throw new ArgumentNullException(nameof(projectServiceAccessor));
             }
 
-            _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
+            if (_joinableTaskFactory == null)
             {
-                var projectServiceAccessor = componentModel.GetService<IProjectServiceAccessor>();
-
+                _joinableTaskFactory = new Lazy<JoinableTaskFactory>(() =>
```

Comment:
You are relying on it always being called from those specific methods in that specific way then. It would be very easy for someone to use this method the wrong way in the future then, especially since it is a public static.